### ISSUE TITLE
[Bug][Move-prover][Compiler] Wrong number of result variables in call to $1_vector_length'$1_code_PackageMetadata'

### ISSUE THREAD

**Issue by rahxephon89** [aptos-labs/aptos-core#8752]
# 🐛 Bug

## To reproduce

After enabling verification for `publish_package` in the `code` module:
```rust
    spec publish_package(owner: &signer, pack: PackageMetadata, code: vector<vector<u8>>) {
        pragma verify = true;
    }
```

The following internal error is raised:

```
{
  "Error": "Move Prover failed: [internal] boogie exited with compilation errors:\n/Users/.../aptos-core/aptos-move/framework/aptos-framework/boogie.bpl(11761,4): Error: wrong number of result variables in call to $1_vector_length'$1_code_PackageMetadata': 2\n/Users/.../aptos-core/aptos-move/framework/aptos-framework/boogie.bpl(11907,4): Error: wrong number of result variables in call to $1_vector_borrow'$1_code_PackageMetadata': 2\n2 name resolution errors detected in /Users/.../aptos-core/aptos-move/framework/aptos-framework/boogie.bpl\n"
}

```

## Expected Behavior

No error is generated

**Comment by rahxephon89** [comment]
Here is a minimal example that triggers the bug: 
```
module 0x42::vec_len {
    use std::vector;
    public fun test(packages: &mut vector<u64>) {
        vector::enumerate_ref(packages, |i, old| {
        });
    }
}
```

**Comment by rahxephon89** [comment]
Fixed by #9954
